CRM/Contribute/BAO/Contribution.php

```diff
@@ -2498,10 +2497,7 @@ public function composeMessageArray(&$input, &$ids, &$values, $returnMessageText
     }
 
     $this->loadRelatedObjects($paymentProcessorID, $ids);
-
-    if (empty($this->_component)) {
-      $this->_component = $input['component'] ?? NULL;
-    }
+    $paymentProcessor = $this->_relatedObjects['paymentProcessor'];
```

Comment:
@mattwire I am suspecting it is this code that is causing this test failure

```
api_v3_OrderTest::testAddOrderForMembership
Failure in api call for Payment create:  Undefined array key "paymentProcessor"
```

Reply:
Thanks @seamuslee001 yes I think you're right